Tests/NIOCertificateReloadingTests/TimedCertificateReloaderTests.swift

```diff
@@ -484,6 +530,42 @@ final class TimedCertificateReloaderTests: XCTestCase {
             issuerPrivateKey: .init(samplePrivateKey)
         )
     }()
+    static let sampleCertChain: [Certificate] = {
+        [
+            try! Certificate(
+                version: .v3,
+                serialNumber: .init(),
+                publicKey: .init(samplePrivateKey.publicKey),
+                notValidBefore: startDate.advanced(by: -60 * 60 * 24 * 360),
+                notValidAfter: startDate.advanced(by: 60 * 60 * 24 * 360),
+                issuer: sampleCertName,
+                subject: sampleCertName,
+                signatureAlgorithm: .ecdsaWithSHA384,
+                extensions: Certificate.Extensions {
+                    Critical(
+                        BasicConstraints.isCertificateAuthority(maxPathLength: nil)
+                    )
+                },
+                issuerPrivateKey: .init(samplePrivateKey)
+            ),
+            try! Certificate(
+                version: .v3,
+                serialNumber: .init(),
+                publicKey: .init(samplePrivateKey.publicKey),
+                notValidBefore: startDate.advanced(by: -60 * 60 * 24 * 360),
+                notValidAfter: startDate.advanced(by: 60 * 60 * 24 * 360),
+                issuer: sampleCertName,
+                subject: sampleCertName,
+                signatureAlgorithm: .ecdsaWithSHA384,
+                extensions: Certificate.Extensions {
+                    Critical(
+                        BasicConstraints.isCertificateAuthority(maxPathLength: nil)
+                    )
+                },
+                issuerPrivateKey: .init(samplePrivateKey)
```

Comment:
Could you use a different private key here? I want to make sure we test that we're validating the key against the right cert (the first one).